### PR TITLE
Add book download

### PR DESCRIPTION
**Pull request by gestchild**
References #4228

- Adds PDF download option for books
- Instead of passing a iiif image url to the Download component we now pass an array of download options
- The download options come from the iiifPresentationManifest if we have one, else they are generate from the iiif imageUrl if we have that
- N.B. this does mean making a second API call for the manifest, but can't see a way around that
(but since we're doing this we should send if down the pipe, then the IIIFPresentationPreview won't need to request it again - will do this in separate PR)

Screen Shot 2019-03-12 at 16 58 38

Next PR: 
- getting the appropriate licensing information
- maybe using context for the download props?

### CODE REVIEW

File: catalogue/webapp/components/Download/Download.js [hunk 1]
```diff
@@ -140,69 +140,58 @@ const Download = ({
...
-                </span>
-              </a>
-            </li>
-          </ul>
+          {downloadOptions && (
+            <ul className="plain-list no-margin no-padding">
+              {downloadOptions
+                .filter(option => option.format !== 'text/plain') // We're taking out raw text for now
+                .map(option => {
+                  // Doing this for the action so analytics is constant, speak to Hayley about removing this
```
Comment (jamesgorrie) [inline catalogue/webapp/components/Download/Download.js]: `TODO: speak to Hayley...`?

File: catalogue/webapp/components/Download/Download.js [hunk 2]
```diff
@@ -140,69 +140,58 @@ const Download = ({
...
+                  const action =
+                    option.label === 'Download full size'
+                      ? 'download large work image'
+                      : option.label === 'Download small (760px)'
+                      ? 'download small work image'
+                      : option.label;
+                  const format = option.format.split('/')[1].toUpperCase();
```
Comment (jamesgorrie) [inline catalogue/webapp/components/Download/Download.js]: Are we always going to have a `/`? 
We could be a little more defensive here.

File: catalogue/webapp/pages/work.js [hunk 3]
```diff
@@ -222,7 +237,18 @@ WorkPage.getInitialProps = async (
...
 
+  let iiifPresentationDownloadOptions = [];
+  if (iiifPresentationLocation) {
+    try {
+      const iiifManifest = await fetch(iiifPresentationLocation.url);
+      const manifestData = await iiifManifest.json();
+      iiifPresentationDownloadOptions = getDownloadOptionsFromManifest(
+        manifestData
+      );
+    } catch (e) {}
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/work.js]: A little worried about relying on on the DCLS stuff here.
We could do it client side, and only when someone clicks to the button to save on bandwidth.

I wonder if there is a very lo-fi version where we just render the links on the screen at `/works/{id}/downloads`?

Is there a reason to do parsing of the options in the getInitialProps over doing it in the component?

File: common/views/components/Tags/Tags.js [hunk 4]
```diff
@@ -33,10 +33,10 @@ const Tags = ({ tags }: Props) => {
           [spacing({ s: 0 }, { padding: ['left'], margin: ['top'] })]: true,
         })}
       >
-        {tags.map(({ textParts, linkAttributes }) => {
+        {tags.map(({ textParts, linkAttributes }, i) => {
           return (
             <li
-              key={textParts.join('-')}
+              key={textParts.concat(i).join('-')}
```
Comment (jamesgorrie) [inline common/views/components/Tags/Tags.js]: Could we use just `i`?

**Comment by gestchild** [comment]
closing in favour of #4255

**Pull request closed without merging**